### ISSUE TITLE
recent homebrew changes: python points to python3

### ISSUE THREAD

**Issue by boblannon** [mapbox/node-pre-gyp#352]
Opening this ticket as an FYI to maintainers, and a heads-up to anyone who might be confused by seeing this error.  Also hoping to get some tips about how best to work around.

Ran into a build error on carmen-cache, related to python versioning:

```
$ make
V=1 ./node_modules/.bin/node-pre-gyp configure build --loglevel=error
gyp ERR! configure error
gyp ERR! stack Error: Python executable "/usr/local/bin/python" is v3.6.4, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
gyp ERR! stack     at Object.failPythonVersion 
```

This is because [recent homebrew changes](https://brew.sh/2018/01/19/homebrew-1.5.0/) make it so that `python` points to 3.x instead of 2, as it did before:

> On 1st March 2018 the python formula will be upgraded to Python 3.x and a python@2 formula will be added for installing Python 2.7 

As the error message says, the `--python` switch can be used to point to the right Python executable, but it might be tricky to set this in a way that's consistent across environments.

**Comment by springmeyer** [comment]
Thanks for ticketing and for the heads up @boblannon. I agree that requiring a flag to be passed manually could be tricky, so making this somehow automatic would be ideal. But how to make it automatic and robust...

**Comment by boblannon** [comment]
> But how to make it automatic and robust...

yep, no ideas there unfortunately.  I'm sure this headache is shared with many :/

**Comment by boblannon** [comment]
just to follow up on this with a potential workaround. assuming you've upgraded homebrew (so that `python` now points to 3.6)...

1. install the python2 keg:
```
brew install python@2
```

2.  make sure it's properly linked
```
brew link python@2
````

3. get the absolute path to the python2.7 binary
```
which python2.7
```

4. use the `PYTHON` env variable when executing code that uses node-pre-gyp, eg:
```
PYTHON=/usr/local/bin/python2.7 make build
```

**Comment by rammygit** [comment]
thank you @boblannon , this workaround worked for node-gyp source build when calling from npm
